Windows graphics changes

shoustech opened the pull request:
## Note:
I am not using the previously used variable `ssections` because I could not find a reliable way to map any of the displays to the values output. I can go back in and add `pixelDepth` and `main` with the value from the main display in `ssections` as the default for all the monitors. 
## Description
I am opening this pull request because after #853 was submitted I ended up looking at the commands being used to pull the information on windows and noticed that there was some room for improvement. I fixed an issue where in some situations the output would exclude displays. I have also added the fields `displayId` and `productionYear` which was previously only available on MacOS.

## Notable changes
The number of displays is now based on the variable `tsections`. I then match the InstanceName property to the Instance name property in `msections` and `isections`. For `dsections` I am finding the correct display by matching `tsections.InstanceName` to `dsections.PNPDeviceID`

After all of the objects associated with the display have been found the app will then pull variables in this order:
`tsection`
`dsection`
`msection`
and finally `isection`

Since the output of `dsection` is the most complete I pulled what I could from that object first and then went down the list from there. If a variable was not pulled from `dsection` or the correlated object could not be found I would then fill in blank values or pull backup variables from the other objects.

Also:
- Added the fields: `displayId` and `productionYear`

## Examples
 In this situation I am using a RDP connection to a different desktop

Existing module
```JSON
    "displays": [
      {
        "vendor": "",
        "model": "",
        "deviceName": "\\\\.\\DISPLAY17",
        "main": true,
        "builtin": false,
        "connection": "DVI",
        "resolutionX": 2560,
        "resolutionY": 1440,
        "sizeX": 53,
        "sizeY": 30,
        "pixelDepth": 32,
        "currentResX": 2560,
        "currentResY": 1440,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 84
      }
    ]
```
Modified module
```JSON
    "displays": [
      {
        "vendor": "Dell Inc.",
        "deviceName": "Dell E2414H (Digital - DVI)",
        "model": "Dell E2414H",
        "productionYear": 2014,
        "serial": "VMT444ASU",
        "displayId": "DISPLAY\\DEL4091\\5&21dbb09b&0&UID24832_0",
        "main": "",
        "builtin": false,
        "connection": "DVI",
        "sizeX": 53,
        "sizeY": 30,
        "pixelDepth": 0,
        "resolutionX": 2560,
        "resolutionY": 1440,
        "currentResX": 2560,
        "currentResY": 1440,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 0
      },
      {
        "vendor": "D E L",
        "deviceName": "DELL U719D",
        "model": "DELL U719D",
        "productionYear": 2018,
        "serial": "GLQW7R",
        "displayId": "DISPLAY\\DEL415F\\5&21dbb09b&0&UID24833_0",
        "main": "",
        "builtin": false,
        "connection": "HDMI",
        "sizeX": 60,
        "sizeY": 34,
        "pixelDepth": 0,
        "resolutionX": 0,
        "resolutionY": 0,
        "currentResX": 0,
        "currentResY": 0,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 0
      },
      {
        "vendor": "L E X",
        "deviceName": "A409U",
        "model": "A409U",
        "productionYear": 2015,
        "serial": "16843009",
        "displayId": "DISPLAY\\LEX0000\\5&21dbb09b&0&UID24836_0",
        "main": "",
        "builtin": false,
        "connection": "DP",
        "sizeX": 128,
        "sizeY": 72,
        "pixelDepth": 0,
        "resolutionX": 0,
        "resolutionY": 0,
        "currentResX": 0,
        "currentResY": 0,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 0
      },
      {
        "vendor": "H W",
        "deviceName": "H V41",
        "model": "H V41",
        "productionYear": 2014,
        "serial": "3CQ4113RS",
        "displayId": "DISPLAY\\HWP311F\\5&21dbb09b&0&UID24839_0",
        "main": "",
        "builtin": false,
        "connection": "DVI",
        "sizeX": 52,
        "sizeY": 29,
        "pixelDepth": 0,
        "resolutionX": 0,
        "resolutionY": 0,
        "currentResX": 0,
        "currentResY": 0,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 0
      },
      {
        "vendor": "(Standard monitor types)", //Current display 17
        "deviceName": "Generic Non-PnP Monitor",
        "model": "Generic Non-PnP Monitor",
        "productionYear": 0,
        "displayId": "DISPLAY\\Default_Monitor\\1&31c5ecd4&0&UID256_0",
        "main": "",
        "builtin": true,
        "connection": "RDP",
        "sizeX": 0,
        "sizeY": 0,
        "pixelDepth": 0,
        "resolutionX": 0,
        "resolutionY": 0,
        "currentResX": 0,
        "currentResY": 0,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 0
      }
    ]
```
Without RDP original code:
```JSON
    "displays": [
      {
        "vendor": "Dell Inc.",
        "model": "Dell E2414H (Digital - DVI)",
        "deviceName": "\\\\.\\DISPLAY1",
        "main": false,
        "builtin": false,
        "connection": "DVI",
        "resolutionX": 2560,
        "resolutionY": 1440,
        "sizeX": 53,
        "sizeY": 30,
        "pixelDepth": "32",
        "currentResX": 2560,
        "currentResY": 1440,
        "positionX": -2560,
        "positionY": 740,
        "currentRefreshRate": 84
      },
      {
        "vendor": "",
        "model": "",
        "deviceName": "\\\\.\\DISPLAY2",
        "main": true,
        "builtin": false,
        "connection": "HDMI",
        "resolutionX": 3840,
        "resolutionY": 2160,
        "sizeX": 60,
        "sizeY": 34,
        "pixelDepth": "32",
        "currentResX": 3840,
        "currentResY": 2160,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 84
      },
      {
        "vendor": "",
        "model": "",
        "deviceName": "\\\\.\\DISPLAY3",
        "main": false,
        "builtin": false,
        "connection": "DP",
        "resolutionX": 1920,
        "resolutionY": 1080,
        "sizeX": 128,
        "sizeY": 72,
        "pixelDepth": "32",
        "currentResX": 1920,
        "currentResY": 1080,
        "positionX": 7,
        "positionY": -1080,
        "currentRefreshRate": 84
      },
      {
        "vendor": "",
        "model": "",
        "deviceName": "\\\\.\\DISPLAY4",
        "main": false,
        "builtin": false,
        "connection": "DVI",
        "resolutionX": 1920,
        "resolutionY": 1080,
        "sizeX": 52,
        "sizeY": 29,
        "pixelDepth": "32",
        "currentResX": 1920,
        "currentResY": 1080,
        "positionX": 1927,
        "positionY": -1080,
        "currentRefreshRate": 84
      }
    ]
```
Without RDP modified code:
```JSON
"displays": [
      {
        "vendor": "Dell Inc.",
        "deviceName": "Dell E2414H (Digital - DVI)",
        "model": "Dell E2414H",
        "productionYear": 2014,
        "serial": "VMT444ASU",
        "displayId": "DISPLAY\\DEL4091\\5&21dbb09b&0&UID24832_0",
        "main": "",
        "builtin": false,
        "connection": "DVI",
        "sizeX": 53,
        "sizeY": 30,
        "pixelDepth": 0,
        "resolutionX": 0,
        "resolutionY": 0,
        "currentResX": 0,
        "currentResY": 0,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 0
      },
      {
        "vendor": "D E L",
        "deviceName": "DELL U719D",
        "model": "DELL U719D",
        "productionYear": 2018,
        "serial": "GLQW7R",
        "displayId": "DISPLAY\\DEL415F\\5&21dbb09b&0&UID24833_0",
        "main": "",
        "builtin": false,
        "connection": "HDMI",
        "sizeX": 60,
        "sizeY": 34,
        "pixelDepth": 0,
        "resolutionX": 0,
        "resolutionY": 0,
        "currentResX": 0,
        "currentResY": 0,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 0
      },
      {
        "vendor": "L E X",
        "deviceName": "A409U",
        "model": "A409U",
        "productionYear": 2015,
        "serial": "16843009",
        "displayId": "DISPLAY\\LEX0000\\5&21dbb09b&0&UID24836_0",
        "main": "",
        "builtin": false,
        "connection": "DP",
        "sizeX": 128,
        "sizeY": 72,
        "pixelDepth": 0,
        "resolutionX": 0,
        "resolutionY": 0,
        "currentResX": 0,
        "currentResY": 0,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 0
      },
      {
        "vendor": "H W",
        "deviceName": "H V41",
        "model": "H V41",
        "productionYear": 2014,
        "serial": "3CQ4113RS",
        "displayId": "DISPLAY\\HWP311F\\5&21dbb09b&0&UID24839_0",
        "main": "",
        "builtin": false,
        "connection": "DVI",
        "sizeX": 52,
        "sizeY": 29,
        "pixelDepth": 0,
        "resolutionX": 0,
        "resolutionY": 0,
        "currentResX": 0,
        "currentResY": 0,
        "positionX": 0,
        "positionY": 0,
        "currentRefreshRate": 0
      }
    ]
```

shoustech committed:
Changed windows class the baseboard function was using to pull the SMBIOSAssetTag from.

shoustech committed:
New fields and fixed missing information from windows displays

shoustech committed:
Added missing RDP videoType

shoustech committed:
Completed Windows graphic changes

shoustech committed:
Fixed incorrect value in initial display variable

shoustech committed:
Removed varible that will always be empty. Converted model year to an integer

shoustech committed:
Removed development console.log

sebhildebrandt commented:
@shoustech wow, thank you! I will check it the upcoming days and if is also works here on my side, I will crate a new release.

shoustech committed:
Update graphics.js

Removed unneeded if statements. Added a backup for when we do not recognize the monitor display technology.

shoustech committed:
Update wifi.js

Fixed an issue on mac systems where wifi.js was reporting an error when trying to list networks.

shoustech committed:
Merge pull request #1 from shoustech/bugfix/#851

Bugfix/#851

shoustech committed:
Merge branch 'master' into windows_display_changes

shoustech committed:
Updated systeminformation with changes from master

Joniras commented:
@shoustech i am really interested in the code to find out if bitlocker is set up. i could not find it in your fork, where could i find the code?

shoustech committed:
Fixed network.js vulnerability

shoustech committed:
Update wifi.js

shoustech committed:
Merge pull request #4 from shoustech/shoustech-patch-1

Update wifi.js

sebhildebrandt commented:
@shoustech after a (very) long time (sorry for that) I was able to work on a couple of issues for graphics(). I also tackled your suggestions and I think this should be solved now. Version 5.32.0 just released. Can you test it on your side?
